core/device_impl.cpp

```diff
@@ -443,6 +444,40 @@ void DeviceImpl::get_param_float_async(const std::string &name,
                                             callback));
 }
 
+MavlinkCommands::Result DeviceImpl::set_flight_mode(uint8_t custom_sub_mode, uint8_t custom_mode)
```

Comment:
Lets have `DeviceImpl::FlightMode` similar to `Telemetry::FlightMode`. Telemetry flight mode methods shall uses the services of those in `DeviceImpl`. We should have simple signature of `set_flight_mode` as below:
```
MavlinkCommands::Result DeviceImpl::set_flight_mode(DeviceImpl::FlightMode flight_mode);
```